app/MMB/src/scenes/timeline/events/AirportArrivalTimelineEvent.js

```diff
@@ -15,41 +21,68 @@ import TimelineEventDateLocationContext from '../../../context/TimelineEventDate
 
 type Props = {|
   +data: ?AirportArrivalTimelineEventType,
+  +navigation: NavigationType,
 |};
 
-const AirportArrivalTimelineEvent = (props: Props) => {
-  const airport = idx(props, _ => _.data.location.airport.name);
-  const code = idx(props, _ => _.data.location.airport.locationId) || '';
-  const timestamp = idx(props, _ => _.data.timestamp);
-  return (
-    <TimelineEventDateLocationContext.Provider value={{ highlightText: true }}>
-      <TimelineEvent
-        timestamp={timestamp}
-        place={code}
-        iconVertLines={<TimelineEventIcon icon={<TextIcon code="a" />} />}
-        mainContent={
-          <React.Fragment>
-            <Title>
-              <Translation
-                id="mmb.booking_timeline.event.airport_arrival.title"
-                values={{
-                  airport,
-                  code,
-                }}
-              />
-            </Title>
-            <Note>
-              <Translation id="mmb.booking_timeline.event.airport_arrival.note" />
-            </Note>
-          </React.Fragment>
-        }
-      />
-    </TimelineEventDateLocationContext.Provider>
-  );
-};
+class AirportArrivalTimelineEvent extends React.Component<Props> {
+  navigate = (key: RouteNamesType, params?: Object) => {
+    this.props.navigation.navigate({
+      routeName: key,
+      key: `key-${key}`,
+      params,
+    });
+  };
+
+  handleOpenExplore = () => {
+    this.navigate('mmb.explore', {
+      locationId: idx(this.props.data, _ => _.location.airport.locationId),
+    });
+  };
+
+  render() {
+    const airport = idx(this.props, _ => _.data.location.airport.name);
+    const code = idx(this.props, _ => _.data.location.airport.locationId) || '';
+    const timestamp = idx(this.props, _ => _.data.timestamp);
+    return (
+      <TimelineEventDateLocationContext.Provider
+        value={{ highlightText: true }}
+      >
+        <TimelineEvent
+          timestamp={timestamp}
+          place={code}
+          iconVertLines={<TimelineEventIcon icon={<TextIcon code="a" />} />}
+          mainContent={
+            <React.Fragment>
+              <Title>
+                <Translation
+                  id="mmb.booking_timeline.event.airport_arrival.title"
+                  values={{
+                    airport,
+                    code,
+                  }}
+                />
+              </Title>
+              <Note>
+                <Translation id="mmb.booking_timeline.event.airport_arrival.note" />
+              </Note>
+              <View style={styles.button}>
+                <TextButton
+                  title={
+                    <Translation id="mmb.booking_timeline.explore_airport.button.title" />
+                  }
+                  onPress={this.handleOpenExplore}
+                />
+              </View>
```

Comment:
This seems to be duplicated in both files. How about creating a new component that takes locationId as a prop and handles the navigation?